src/hooks.js

```diff
@@ -0,0 +1,86 @@
+import {useEffect, useLayoutEffect} from 'react'
+import debounce from 'lodash.debounce'
+import countryIso from 'country-iso'
+import countryData from 'country-data'
+
+import drawChart from './chart'
+import getCountryData from './consumer'
+import geolocationStates from './geolocationStates'
+
+const google = window.google
+
+const getCountrySlug = coords => {
+  const code = countryIso.get(coords.latitude, coords.longitude)
+  return countryData.countries[code].name
+}
+
+const setUpGeolocation = async ({coords, setGeolocationState, setSelectedCountries, setData}) => {
+  const slug = getCountrySlug(coords).toLowerCase()
+  console.log(slug)
+  setGeolocationState(geolocationStates.loading)
+  google.charts.setOnLoadCallback(async () => {
+    const localData = await getCountryData(slug)
+    setGeolocationState(geolocationStates.loaded)
+    setSelectedCountries([slug])
+    setData({ [slug]: localData })
+  })
+}
+
+const handleGeoError = async ({code, fallbackCountry, setGeolocationState, setSelectedCountries, setData}) => {
+  setGeolocationState(
+    code === 1
+      ? geolocationStates.disallowed
+      : geolocationStates.failed
+  )
+  if (code === 1) {
+    const fallbackData = await getCountryData(fallbackCountry)
+    setSelectedCountries([fallbackCountry])
+    setData({ [fallbackCountry]: fallbackData })
+  }
+}
+
+const useChartUpdate = ({data, selectedCountries}) => {
+  useLayoutEffect(() => {
+    if (google.visualization) {
+      if (!selectedCountries.some(country => !data[country])) {
+        drawChart({data, selectedCountries})
+      }
+    }
+  }, [data, selectedCountries])
+}
+
+const useResizeListener = ({data, selectedCountries}) => {
```

Comment:
I noticed that the `window.resize` event handler is a bit trigger happy and fails if you resize the window before the application properly loads.

Reply:
True. I added a new condition there. Not sure if it's the best way to approach it but it seems to work.